apps/vercel/app-actions/src/services/vercel-service.ts

```diff
@@ -1,18 +1,29 @@
 import { VercelProject } from '../types';
 
 export class VercelService {
-  constructor(readonly accessToken: string) {}
+  constructor(readonly accessToken: string, readonly teamId: string) {}
 
   public async getProject(projectId: string): Promise<VercelProject> {
     const response = await fetch(this.buildProjectUrl(projectId), {
       method: 'GET',
       headers: this.buildRequestHeaders(),
     });
+    await this.maybeThrowApiError(response);
     const vercelProject = await response.json();
     this.assertVercelProject(vercelProject);
     return vercelProject;
   }
 
+  // cheap and dirty error handling -- could later provide a more structured error response
+  // when API errors are encountered
+  private async maybeThrowApiError(response: Response): Promise<void> {
```

Comment:
this error handling definitely works for now 💯 nit: wondering if the name of this method could best reflect what it is doing, i.e `handleApiError` or `checkResponseStatus`, instead of what the results _potentially_ are? totally fine with the name as is now, just as I was reading the method on line 11, I definitely had to look at the method itself to fully understand what it was doing.

Reply:
It's a good point. I think `handleApiError` is a lot more obvious and close enough to what it does.